core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -340,7 +362,22 @@ private ManifestFile filterManifest(Schema tableSchema, ManifestFile manifest) {
     }
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
This method gets too complicated. What about refactoring it like this?

```
private boolean canContainDeletedFiles(ManifestFile manifest) {
  return canContainDropBySeq(manifest)
      || canContainExpressionDeletes(manifest)
      || canContainDroppedPartitions(manifest)
      || canContainDroppedFiles(manifest);
}
```

Where `canContainDroppedFiles` would contain the new logic (just an example):

```
private boolean canContainDroppedFiles(ManifestFile manifest) {
  if (deleteFileManifests.contains(manifest.path()) || !deletePaths.isEmpty()) {
    return true;
  } else if (allDeletesReferenceManifests && trustReferencedManifests) {
    return false;
  } else {
    return ManifestFileUtil.canContainAny(manifest, deleteFilePartitions, specsById);
  }
}
```

Reply:
Good suggestion! yeah let's just break this method apart instead of adding more cases and suppressing the checkstyle warning